lib/assert.js

```diff
@@ -94,13 +94,21 @@ x.throws = function (fn, err, msg) {
 
 		var result;
 
-		assert.throws(function () {
-			try {
-				fn();
-			} catch (error) {
-				result = error;
-				throw error;
+		try {
+			fn();
+		} catch (error) {
+			if (error && typeof error !== 'object') {
```

Comment:
Not sure if this check is good enough in every case though. Should I use something like [is-error](https://www.npmjs.com/package/is-error) instead? Because now someone can still `throw {foo: 'bar'}`.